packages/loader/driver-utils/src/storageUtils.ts

```diff
@@ -0,0 +1,18 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISnapshot } from "@fluidframework/driver-definitions";
+import { ISnapshotTree } from "@fluidframework/protocol-definitions";
+
+/**
+ * Utility API to check if the type of snapshot contents is `ISnapshot`.
+ * @internal
+ * @param obj - obj whose type needs to be identified.
+ */
+export function isInstanceOfISnapshot(
+	obj: ISnapshotTree | ISnapshot | undefined,
+): obj is ISnapshot {
+	return obj !== undefined && "snapshotInNewFormat" in obj && obj.snapshotInNewFormat === true;
```

Comment:
I think you can remove ""snapshotInNewFormat" in obj" check - it would be undefined if property is not there.
You can even shorten the whole thing to
```ts
obj?.snapshotInNewFormat === true
```

Reply:
No, we can not do this because "snapshotInNewFormat" prop does not exist in type iSnapshotTree. The compiler complains. We need to use the in operator.